### PR TITLE
Upgrade terraform-provider-vault to v3.9.1

### LINKED ISSUE #186: Upgrade terraform-provider-vault to v3.9.1
Release details: https://github.com/hashicorp/terraform-provider-vault/releases/tag/v3.9.1

### LINKED ISSUE #189: Refreshing an Identity Group that does not exist causes an error
### What happened?

When attempting to refresh an Identity Group that does not exist, it indeed establishes that it can't be found - but instead of removing it from the stack, it errors out.

### Steps to reproduce

- Create an Identity Group via Pulumi
- Delete the Identity Group manually (via Vault GUI or CLI)
- Run `pulumi refresh` or `pulumi up --refresh`

### Expected Behavior

Pulumi should identify that the resources don't exist.

### Actual Behavior

Error:

```
Diagnostics:
  vault:identity:Group (<REDACTED>):
    error: Preview failed: refreshing urn:pulumi:devops::vault-namespace::vault:identity/group:Group::<REDACTED>: 1 error occurred:
    	* error checking if IdentityGroup "<REDACTED>" exists: entity not found: "/identity/group/id/86ab3738-1226-43c0-1c37-22a8c4e38464"

  pulumi:pulumi:Stack (vault-namespace-devops):
    error: preview failed

  vault:identity:Group (<REDACTED>):
    error: Preview failed: refreshing urn:pulumi:devops::vault-namespace::vault:identity/group:Group::<REDACTED>: 1 error occurred:
    	* error checking if IdentityGroup "<REDACTED>" exists: entity not found: "/identity/group/id/4bd58193-d8c4-3384-9b9b-1771cfcdedd5"
```

### Output of `pulumi about`

```
CLI
Version      3.42.0
Go Version   go1.19.1
Go Compiler  gc

Plugins
NAME    VERSION
nodejs  unknown
vault   5.6.0

Host
OS       darwin
Version  12.5
Arch     x86_64

This project is written in nodejs: executable='/usr/local/bin/node' version='v18.9.1'

Backend
Name           <REDACTED>
URL            S3 <REDACTED>
User           <REDACTED>
Organizations

Dependencies:
NAME            VERSION
@pulumi/pulumi  3.42.0
@pulumi/vault   5.6.0
@types/node     14.18.31
```

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by viveklak**
Fixes #189 
Fixes #186

### CODE REVIEW

File: provider/resources.go [hunk 1]
```diff
@@ -159,7 +161,10 @@ func Provider() tfbridge.ProviderInfo {
 			"vault_mfa_pingid":             {Tok: makeResource(mainMod, "MfaPingid")},
 			"vault_mfa_totp":               {Tok: makeResource(mainMod, "MfaTotp")},
 			"vault_mount":                  {Tok: makeResource(mainMod, "Mount")},
-			"vault_namespace":              {Tok: makeResource(mainMod, "Namespace")},
+			"vault_namespace": {Tok: makeResource(mainMod, "Namespace"),
+				Fields: map[string]*tfbridge.SchemaInfo{"namespace": {
+					CSharpName: "TargetNamespace", // error CS0542: 'Namespace': member names cannot be the same as their enclosing type
```
Comment (viveklak) [inline provider/resources.go:166]: Had to hack this to avoid dotnet SDK build failure.

File: provider/resources.go [hunk 2]
```diff
@@ -288,6 +294,11 @@ func Provider() tfbridge.ProviderInfo {
 					Source: "database_secret_backend_role.md",
 				},
 			},
+			"vault_identity_mfa_duo":                    {Tok: makeResource(identityMod, "MfaDuo")},
+			"vault_identity_mfa_login_enforcement":      {Tok: makeResource(identityMod, "MfaLoginEnforcement")},
+			"vault_identity_mfa_okta":                   {Tok: makeResource(identityMod, "MfaOkta")},
+			"vault_identity_mfa_pingid":                 {Tok: makeResource(identityMod, "MfaPingID")},
```
Comment (guineveresaenger) [inline provider/resources.go:300]: by convention we do capitalize only after the `_` character, so we should use
`"MfaPingid"`

File: provider/resources.go [hunk 3]
```diff
@@ -288,6 +294,11 @@ func Provider() tfbridge.ProviderInfo {
...
+			"vault_identity_mfa_totp":                   {Tok: makeResource(identityMod, "MfaTOTP")},
```
Comment (guineveresaenger) [inline provider/resources.go:301]: Similar to above:
```suggestion
			"vault_identity_mfa_totp":                   {Tok: makeResource(identityMod, "MfaTotp")},
```

File: provider/resources.go [hunk 4]
```diff
@@ -460,6 +485,14 @@ func Provider() tfbridge.ProviderInfo {
 					Source: "kubernetes_auth_backend_role.md",
 				},
 			},
+			"vault_kubernetes_service_account_token": {Tok: makeDataSource(kubernetesMod, "getServiceAccountToken")},
+
+			// KV
+			"vault_kv_secret":            {Tok: makeDataSource(kvMod, "getSecret")},
+			"vault_kv_secret_subkeys_v2": {Tok: makeDataSource(kvMod, "getSecretSubKeysV2")},
```
Comment (guineveresaenger) [inline provider/resources.go:492]:
```suggestion
			"vault_kv_secret_subkeys_v2": {Tok: makeDataSource(kvMod, "getSecretSubkeysV2")},
```

File: provider/resources.go [hunk 5]
```diff
@@ -460,6 +485,14 @@ func Provider() tfbridge.ProviderInfo {
...
+			"vault_kv_secret_v2":         {Tok: makeDataSource(kvMod, "getSecretV2")},
+			"vault_kv_secrets_list":      {Tok: makeDataSource(kvMod, "getSecretsList")},
+			"vault_kv_secrets_list_v2":   {Tok: makeDataSource(kvMod, "getSecretsV2List")},
```
Comment (guineveresaenger) [inline provider/resources.go:495]:
```suggestion
			"vault_kv_secrets_list_v2":   {Tok: makeDataSource(kvMod, "getSecretsListV2")},
```

**Review by guineveresaenger — CHANGES_REQUESTED** [review]
thank you for cutting this upgrade! a few twiddly bits with naming convention but the breaking changes LGTM.

**Comment by viveklak** [comment]
> thank you for cutting this upgrade! a few twiddly bits with naming convention but the breaking changes LGTM.

Thanks for catching! I think I addressed all your requests. Let me know if I missed something.

**Comment by guineveresaenger** [comment]
yes, this looks good to me ! thank you for the followup fix!

**Pull request merged**